projects/ghostscript/gstoraster_fuzzer.cc

```diff
@@ -110,7 +111,39 @@ static int gs_to_raster_fuzz(const unsigned char *buf, size_t size)
 	return 0;
 }
 
+#ifdef PDF_TARGET
+// Returns 1 if this has a valid PDF header and 0 otherwise
+static int quick_check_pdf(const uint8_t *data, size_t size) {
+	// PDF checks. Exit early if we don't have a valid PDF signature.
+	if (size < 5) {
+		return 0;
+	}
+
+	// Check PDF tag. We do this because we want to use seeds
+	if (data[0] != 0x25 || data[1] != 0x50 || data[2] != 0x44 || data[3] != 0x46 || data[4] != 0x2d) {
```

Comment:
Is this primarily a speed optimization?  Ghostscript users somewhat rely on magic byte checking.

Reply:
It's for two reasons: optimisation and narrowing the scope of the fuzzer. I would like to have a fuzzer that purely explores PDF code and we can approximate this check. This is partly inspired by the v8 fuzzer here: https://github.com/v8/v8/blob/a8fad3884dfdf0bcce1cfed053d84c44839bc363/test/fuzzer/parser.cc#L23

Reply:
That makes sense.  There's not a lot of value in trying to fuzz both PostScript and PDF at the same time.